girder-dandi-archive/tests/test_rest.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+
+from girder_dandi_archive.util import get_or_create_drafts_collection
+
+from girder.models.collection import Collection
+from girder.models.folder import Folder
+
+from pytest_girder.assertions import assertStatus, assertStatusOk
+
+
+# TODO: This fixture and these tests are not well structured and hard to maintain.
+# They currently work correctly in terms of the application code that they test,
+# but they should be rewritten and not followed as an example.
+@pytest.fixture
+def draftsFolders(db):
+    red_herring_collection = Collection().createCollection(
+        "red_herring_collection", reuseExisting=True
+    )
+    # A folder that matches dandiset metadata, but not in drafts collection.
+    red_herring_dandiset_000000_folder = Folder().createFolder(
+        parent=red_herring_collection,
+        parentType="collection",
+        name="000000",
+        public=True,
+    )
+    meta = {
+        "dandiset": {"name": "red", "description": "herring", "identifier": "000000"}
+    }
+    Folder().setMetadata(red_herring_dandiset_000000_folder, meta)
+    return_dict = {
+        "red_herring_collection": red_herring_collection,
+        "red_herring_folder": red_herring_dandiset_000000_folder,
+    }
+    return_dict["drafts_collection"] = get_or_create_drafts_collection()
+    yield return_dict
```

Comment:
To make the syntax simpler and following what Brian suggested:

```suggestion
    return_dict = {
        "red_herring_collection": red_herring_collection,
        "red_herring_folder": red_herring_dandiset_000000_folder,
        "drafts_collection": get_or_create_drafts_collection(),
    }
    return return_dict
```

Reply:
I'll address this with #126, I'm doing all of the test restructuring there so that I can get this PR in sooner.